src/core/src/main/kotlin/org/apache/jmeter/testelement/TestElement.kt

```diff
@@ -0,0 +1,269 @@
+     * the name of the property to get
+     * @return the value of the property
+     */
+    public fun getPropertyAsDouble(key: String): Double
+
+    /**
+     * Tells the test element to return to the state it was in when
+     * setRunningVersion(true) was called.
+     */
+    public fun recoverRunningVersion()
+
+    /**
+     * Clear the TestElement of all data.
+     */
+    public fun clear()
+
+    /**
+     * Return a property as a string value.
+     *
+     * @param key the name of the property to get
+     * @return the value of the property
+     */
+    public fun getPropertyAsString(key: String): String
+
+    /**
+     * Return a property as an string value or a default value if no property
+     * could be found.
+     *
+     * @param key
+     * the name of the property to get
+     * @param defaultValue
+     * the default value to use
+     * @return the value of the property, or `defaultValue` if no
+     * property could be found
+     */
+    public fun getPropertyAsString(key: String, defaultValue: String): String
+
+    /**
+     * Sets and overwrites a property in the TestElement. This call will be
+     * ignored if the TestElement is currently a "running version".
+     *
+     * @param property the property to be set
+     */
+    public fun setProperty(property: JMeterProperty)
+
+    /**
+     * Given the name of the property, returns the appropriate property from
+     * JMeter. If it is null, a [NullProperty] object will be returned.
+     *
+     * @param propName the name of the property to get
+     * @return [JMeterProperty] stored under the name, or [NullProperty] if no property can be found
+     */
+    public fun getProperty(propName: String): JMeterProperty
+
+    /**
+     * Get a Property Iterator for the TestElements properties.
+     *
+     * @return PropertyIterator
+     */
+    public fun propertyIterator(): PropertyIterator
+
+    /**
+     * Remove property stored under the `key`
+     *
+     * @param key name of the property to be removed
+     */
+    public fun removeProperty(key: String)
+
+    // lifecycle methods
+    public override fun clone(): Any
+
+    /**
+     * Convenient way to traverse a test element.
+     *
+     * @param traverser The traverser that is notified of the contained elements
+     */
+    public fun traverse(traverser: TestElementTraverser)
+
+    /**
+     * Associates a thread context with this element.
+     */
+    public var threadContext: JMeterContext
+
+    /**
+     * Associates a thread name with this element.
+     */
+    public var threadName: String
+
+    /**
+     * Called by Remove to determine if it is safe to remove the element. The
+     * element can either clean itself up, and return true, or the element can
+     * return false.
+     *
+     * @return true if safe to remove the element
+     */
+    public fun canRemove(): Boolean
+
+    /**
+     * Associates a name with this element.
+     */
+    public var name: String?
```

Comment:
I'm not sure if `name` should be nullable, however, I keep it for backward compatibility